_dev/coding-style.md

```diff
@@ -6,8 +6,65 @@ description: The preferred style for code committed to NeoMutt
 
 # Coding Style
 
-Every project has its own coding style, this one is no exception.
-However, I haven't written down the rules.  For now, ...
+## at the moment (pre-fork)
 
-- Trailing whitespace is **evil**
+Generally speaking, we don't want to diverge much from upstream mutt. We're
+still merging patches in - and if we didn't follow upstream's coding style,
+merging would be much harder.
 
+That means:
+
+* put your the braces of functions in new lines, like so:
+
+```c
+struct q_class_t
+{
+  int length;
+  int index;
+  int color;
+  char *prefix;
+  struct q_class_t *next, *prev;
+  struct q_class_t *down, *up;
+};
+```
+
+* define functions with underscores like `void mutt_account_unsetpass (..)`
+
+* what a function does *should* be clear without the need to look into ites
+  implementation - that means use expressive names!
+
+* No global variable is a good variable - therefore avoid them.
+
+* Explain with comments what a function does. That includes:
+
+    - the return value and their meaning
+
+    - parameters and their meaning
```

Comment:
I believe we can be a bit more relaxed about that. IMHO, a good function is a function that is self-explainatory, meaning that you can understand what it does by reading it. And you do that by choosing good parameter names, variable names and function name.

Only the public API, and non obvious functions (we're in C, those can't be avoided) shall be really documented. Inline comments are great too.